lambda/executor.js

```diff
@@ -116,14 +139,17 @@ const runInParallel = async(num, lambdaARN, lambdaAlias, payloads, preARN, postA
         if (invocationResults.FunctionError) {
             throw new Error(`Invocation error (running in parallel): ${invocationResults.Payload} with payload ${JSON.stringify(actualPayload)}`);
         }
+        if (sleepBetweenRunsMs > 0) {
+            await utils.sleep(sleepBetweenRunsMs);
```

Comment:
@ryancormack I've realized this doesn't do what we need :) it works fine for `runInSeries` because the execution of linear, but not for `runInParallel`.

In this case, all the promises are invoked below at:
```javascript
await Promise.all(invocations);
```

so everything will still run in parallel and all promises will wait `sleepBetweenRunsMs` milliseconds, and then invoke the Lambda function at at once.

If we want to implement the intended behavior, we could manipulate the actual value of `sleepBetweenRunsMs` to make sure each promise waits for a different amount of ms.

I could look something like this:
```javascript
const sleeBetweenRunsMsInParallel = sleepBetweenRunsMs * i; // multiply by the promise index
if (sleeBetweenRunsMsInParallel > 0) {
    await utils.sleep(sleeBetweenRunsMsInParallel);
}
```

(I haven't tested this but it looks like the simplest solution, without messing with promises hell)

Reply:
👍🏻 Good spot. I'll take a look at this and give it a go. I've been using `runInSeries` with this for the last few weeks.